astropy/table/table.py

```diff
@@ -765,17 +766,16 @@ def show_in_browser(self,
         path = os.path.join(tmpdir, 'table.html')
 
         with open(path, 'w') as tmp:
-
             if jsviewer:
                 self.write(tmp, format='jsviewer', css=css, max_lines=max_lines,
                            jskwargs=jskwargs, table_id=tableid)
             else:
                 self.write(tmp, format='html')
 
-            if browser == 'default':
-                webbrowser.open("file://" + path)
-            else:
-                webbrowser.get(browser).open("file://" + path)
+        if browser == 'default':
+            webbrowser.open("file://" + path)
+        else:
+            webbrowser.get(browser).open("file://" + path)
```

Comment:
This makes sense--depending on the OS particulars, the output to the file object may not be flushed until it's closed.  This could have added a manual `tmp.flush()` before `webbrowser.open()`, but there's no reason not to just do it like this too.